modules/gdscript/gdscript_parser.cpp

```diff
@@ -1817,7 +1817,7 @@ GDScriptParser::ForNode *GDScriptParser::parse_for() {
 	n_for->list = parse_expression(false);
 
 	if (!n_for->list) {
-		push_error(R"(Expected a list or range after "in".)");
+		push_error(R"(Expected iterable after "in".)");
```

Comment:
Can use an article.

Reply:
Nah, we always use this sort of phrasing for errors in the parser.

Reply:
This bit literally used an article before your edit, and so do other messages that you've adjusted in this PR. We are lazy and lenient, but we definitely don't "always use this" 🙃

Reply:
Fair point, I was in a hurry, so I'll try to reason better now. There are 121 errors that use "expected" like this and only this and another two use an article, which is a really big majority. I think this is deliberate:

- Suppose Godot is talking to you as he shows you the error.
- If he says "Expected an iterable after 'in'", the better interpretation of is "I expected an iterable"
- If he says "Expected iterable after 'in'", the better interpretation is "An iterable is expected".

Thus I think omitting the article conveys a better meaning, as well as being more consistent and shorter.

Reply:
The only reason why it's often written like that is because it's a "computer speech", where the engine talks to you in "tokens". It reads as "Expected <ITERABLE>", which is something that parsers sometimes spew at you. Which is fine for a parser programmer, but it's not great for the end user, who expects to read a humanized error or warning.

But I don't really want to waste time debating this, so I'll just merge it as is. We have plenty of badly phrased error messages in the engine for this to matter.

Reply:
I think the ideal thing would be "Iterable expected after "in"", which is grammatically correct and also feels like Godot isn't personally talking about what he's expecting. I might do it someday, alongside the tall order of making it consistent across the codebase.